operator_csv_libs/imagerepo.py

```diff
@@ -200,6 +202,72 @@ def _get_quay_repo(self):
         r = self.image.get_image_repo().replace('quay.io/','')
         return '/'.join([r, self.image.get_image_name()])
 
+class DockerRepo:
+    
+    def __init__(self,image, docker_user=None, docker_key=None):
+        self.image = image
+        self.org = image.get_image().split('/')[1]
+        self.repo = image.get_image_name()
+        self.tag = image.get_tag()
+
+        if docker_user:
+            self.docker_user = docker_user
+        elif 'DOCKER_USER' in os.environ:
+            # Go to environment variables if we don't have them as class variable either
+            self._docker_user = os.getenv('DOCKER_USER')
+            self.docker_user = self._docker_user
+        else:
+            self.docker_user = None
+
+        if docker_key:
+            self.docker_key = docker_key
+        elif 'DOCKER_KEY' in os.environ:
+            self._docker_key = os.getenv('DOCKER_KEY')
+            self.docker_key = self._docker_key
+        else:
+            self.docker_key = None
+
+    def get_image_digest(self):
+        return self._get_digest(manifest_list=False)
+
+    def get_manifest_list_digest(self):
+        return self._get_digest(manifest_list=True)
+
+    def get_raw_manifest_list(self):
+        ## Get token
+        t=requests.get('https://auth.docker.io/token?scope=repository%3A{org}%2F{repo}%3Apull&service=registry.docker.io'.format(org=self.org, repo=self.repo))
+        token=t.json()['token']
+
+        ## check media type
+        headers={'accept':'application/vnd.docker.distribution.manifest.list.v2+json', 'Authorization': 'Bearer {}'.format(token)}
+        m=requests.get('https://registry-1.docker.io/v2/{org}/{repo}/manifests/{tag}'.format(org=self.org,repo=self.repo,tag=self.tag), headers=headers)
+        
+        if 'manifest.list' in m.headers['Content-Type']:
+            return m.json()
+        else:
+            raise ManifestListNotFound('No manifest for: ' + self.image.get_image())
```

Comment:
It would be good to include some unit tests to validate that we do get the desired outcomes, i.e. that the right exceptions are raised in the expected places, etc.